flax/linen/module.py

```diff
@@ -788,6 +811,12 @@ def apply(self, variables: VariableDict, *args, rngs: RNGSequences = None,
                treated as mutable: ``bool``: all/no collections are mutable.
                ``str``: The name of a single mutable collection. ``list``: A
                list of names of mutable collections.
+      capture_intermediates: If `True`, captures intermediate return values
+        of all Modules inside the "intermediates" collection. By default only
+        the return value of the `__call__` method is stored. A function can
+        be passed to change the filter behavior. The filter function takes
+        the Module instance and method name and returns a bool indicating
+        whether the output of that method invocation should be stored.
```

Comment:
Can we should an example of how you'd write such a custom filter function? (Maybe show how the default one is implemented?)

Reply:
Adding to howto

Reply:
It's probably best as part of the docstring, as people should be able to understand an API from its docstring. Or maybe add a link to the HOWTO from the docstring otherwise?

Reply:
done